### ISSUE TITLE
`require-id-when-available` does not warn for nested fields within a fragment

### ISSUE THREAD

**Issue by jordan-burnett** [B2o5T/graphql-eslint#964]
**Describe the bug**

When using `require-id-when-available` I do not receive a warning when I do not request an ID for a nested field within a fragment

**To Reproduce**

Schema:

```
type Book {
  id: String!
  name: String!
  author: Author!
}

type Author {
  id: String!
  name: String!
}

```

Document:

```
query GetBook {
    book {
       id
       name
        ...BookFragment
    }
}

fragment BookFragment on Book {
   author {
        name
    }
}
```

**Expected behavior**

I would expect to recieve a warning that I did not request `Author.id` in `BookFragment`

**Environment:**

- OS: OSX 
- `@graphql-eslint/3.8.0`
- NodeJS: 14.18.1

**Comment by B2o5T** [comment]
Hi @jordan-burnett, thanks, I know about this issue, even added [commented test](https://github.com/B2o5T/graphql-eslint/blob/master/packages/plugin/tests/require-id-when-available.spec.ts#L277-L286), I will fix it when I have time

**Comment by B2o5T** [comment]
fixed in [@graphql-eslint/eslint-plugin@3.9.0](https://github.com/B2o5T/graphql-eslint/releases/tag/%40graphql-eslint%2Feslint-plugin%403.9.0) 🚀
